instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -164,10 +162,15 @@ class OpenTelemetryMiddleware:
 
     Args:
         app: The ASGI application callable to forward requests to.
-        span_details_callback: Callback which should return a string
-            and a tuple, representing the desired span name and a
-            dictionary with any additional span attributes to set.
-            Optional: Defaults to get_default_span_details.
+        default_span_details: Callback which should return a string and a tuple, representing the desired default span name and a
```

Comment:
Is the default span details callback still needed if we already have `server_request_hook`?

Reply:
Yup see below. I've attempted to utilize the `server_request_hook` to do the reconfiguration, but this would only result in the server span having the correct span name and none of the client/response spans.